pc/manager.cpp

```diff
@@ -746,6 +746,14 @@ void manager::on_response( rpc::get_slot *res )
   if (
     has_status( PC_PYTH_RPC_CONNECTED )
   ) {
+    
+      // New slot received, so flush all pending updates for all active users
+      for( user *uptr = olist_.first(); uptr; ) {
+        user *nptr = uptr->get_next();
+        uptr->send_pending_upds();
+        uptr = nptr;
```

Comment:
why is the loop logic not part of the for loop?

Reply:
another impl option is to have user receive the new slot callback? what's reasoning behind this impl decision?

Reply:
We want to preserve the existing API so that we can quickly roll these changes out to publishers.

Reply:
That iteration pattern was used elsewhere in the codebase so I adopted it for consistency - but yes it could definitely be made simpler.